src/util/optional.h

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************\
```

Comment:
I'm not sure I understand the purpose of this file?

Reply:
Description updated.